README-3PP-LICENSE-CHECK-AND-AUDIT.md

```diff
@@ -0,0 +1,121 @@
+# VS Code builtin extensions: 3PP license check and security vulnerability audit
+
+Note: Further automation is possible for the future, but for now we will document how to do most of this semi-manually (tools are used but could be better integrated in this repo). For example we have a better integration for `dash-licenses` in the main Theia repo, and intend to make it re-usable. When that happens, we can use it in this repo, too.
+
+## "internal" builtin vscode extensions
+
+These are normally just called `vscode builtin extensions`. These are in the vscode git repository, folder `extensions`, built and packaged along `vscode`. They are not made available individually by Microsoft, so to use them with Theia, we need to this ourselves. This is what this repository is for. We have the `vscode` repository as a git submodule. We hook-into its build scripts to build the extensions and then package them individually as `.vsix` files, and publish them to [open-vsx.org](https://open-vsx.org/).
+
+### 3PP License check (builtins)
+
+We use [dash-licenses](https://github.com/eclipse/dash-licenses) to scan for 3rd party dependencies that have unclear or forbidden licenses (as per the Eclipse Foundation rules). It needs to run on each extension.
+
+```bash
+# dash-licenses - local run
+
+yarn download:dash-licenses
+export DASH_LICENSES_JAR=${PWD}/dash-licenses.jar
+# supply your own token. See: https://github.com/eclipse/dash-licenses#automatic-ip-team-review-requests
+export DASH_LICENSES_TOKEN=<token>
+
+# Run dash-licenses on each repo - a summary.txt file will be created, 
+# that contains the results
+find vscode/extensions -maxdepth 1 -mindepth 1 -type d -exec bash -c "cd '{}' &&  pwd && java -jar $DASH_LICENSES_JAR yarn.lock -timeout 120 -batch 20 -summary ./summary.txt" \;
```

Comment:
Good catch. I will come-back to this one after taking care of the easier-to-fix feedback.

Reply:
I have sub-projects covered for the internal built-ins. It found one extra dependency that needed approval : https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/4694

Reply:
I did a similar change for external built-ins. We should now cover better sub-projects (deeper that repo root) for both types of built-ins.

Reply:
This allowed to find a few more dependencies to vet, for external  built-ins:
![image](https://user-images.githubusercontent.com/25749063/200678602-ab0d6dc9-a31f-48f0-8bcd-1ebeb44f4f23.png)

Reply:
FYI, instead of relying on a folder depth, the new way relies on finding `yarn.lock` and/or `package-lock.json` files, no matter the depth. The tools we use (yarn/npm audit, dash-licenses) require one of these to work, so I think we cover as much as possible with this strategy.